flink-core/src/main/java/org/apache/flink/api/common/io/GenericCsvInputFormat.java

```diff
@@ -445,7 +445,12 @@ protected int skipFields(byte[] bytes, int startPos, int limit, byte[] delim) {
 			// quoted string parsing enabled and field is quoted
 			// search for ending quote character
 			i++;
-			while(i < limit && bytes[i] != quoteCharacter) {
+
+			while(i < limit){
+				//quote character was escaped
+				if (bytes[i]==quoteCharacter && bytes[i-1]!=92){
```

Comment:
Can you define a constant `private static final byte BACKSLASH = 92` ? The compiler inlines those, and the code is more self explaining then.